### PR TITLE
Adding support for PubSub. Still needs JavaDocs.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Adding support for PubSub. Still needs JavaDocs.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #66 _

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: examples/src/main/java/io/dapr/examples/pubsub/http/Publisher.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+package io.dapr.examples.pubsub.http;
+
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
```
Comment (AndresRoblesMX) [inline examples/src/main/java/io/dapr/examples/pubsub/http/Publisher.java]: are this really needed?

Comment (artursouza) [reply examples/src/main/java/io/dapr/examples/pubsub/http/Publisher.java]: No. It was another test I was playing with :)

File: sdk-springboot/src/main/java/io/dapr/springboot/DaprController.java [hunk 2]
```diff
@@ -45,17 +47,43 @@ public String daprConfig() throws Exception {
...
+  }
+
+  @RequestMapping(method = RequestMethod.PUT, path = "/{name}")
+  public Mono<byte[]> invokeMethodOrTopicViaPut(@PathVariable("name") String name,
+                                          @RequestBody(required = false) byte[] body,
+                                          @RequestHeader Map<String, String> header) {
+    return Dapr.getInstance().handleInvocation(name, body, header);
+  }
+
   @RequestMapping(method = RequestMethod.POST, path = "/actors/{type}/{id}")
```
Comment (AndresRoblesMX) [inline sdk-springboot/src/main/java/io/dapr/springboot/DaprController.java]: latest version of spring provides a @PostMapping annotation, why not used intead?

Comment (artursouza) [reply sdk-springboot/src/main/java/io/dapr/springboot/DaprController.java]: Thanks. Will do.

File: sdk/src/main/java/io/dapr/client/DaprClientHttpAdapter.java [hunk 3]
```diff
@@ -192,21 +174,21 @@
       if (state.getEtag() != null && !state.getEtag().trim().isEmpty()) {
         headers.put(Constants.HEADER_HTTP_ETAG_ID, state.getEtag());
       }
-      String serializedKeyBody = objectSerializer.serialize(state.getKey());
+      String serializedKeyBody = state.getKey();
       serializedKeyBody += getOptionsAsQueryParameter(options);
       if (options.getConsistency() != null && !options.getConsistency().trim().isEmpty()) {
```
Comment (AndresRoblesMX) [inline sdk/src/main/java/io/dapr/client/DaprClientHttpAdapter.java]: This entire IF is no longer needed, all that logic is inside getOptionsAsQueryParameter method

File: sdk/src/main/java/io/dapr/client/DaprClientHttpAdapter.java [hunk 4]
```diff
@@ -192,21 +174,21 @@
       if (state.getEtag() != null && !state.getEtag().trim().isEmpty()) {
         headers.put(Constants.HEADER_HTTP_ETAG_ID, state.getEtag());
       }
-      String serializedKeyBody = objectSerializer.serialize(state.getKey());
+      String serializedKeyBody = state.getKey();
```
Comment (AndresRoblesMX) [inline sdk/src/main/java/io/dapr/client/DaprClientHttpAdapter.java]: if no need to serialize, do we really need this variable anymore?

File: sdk/src/main/java/io/dapr/client/DaprClientHttpAdapter.java (hunk 3 above)
Comment (artursouza) [reply sdk/src/main/java/io/dapr/client/DaprClientHttpAdapter.java]: Ok. Will remove.

File: sdk/src/main/java/io/dapr/client/DaprClientHttpAdapter.java (hunk 4 above)
Comment (artursouza) [reply sdk/src/main/java/io/dapr/client/DaprClientHttpAdapter.java]: I will change it. I did a big refactoring without changing more than needed. Will update.

File: sdk/src/main/java/io/dapr/client/DaprClientBuilder.java [hunk 5]
```diff
@@ -93,15 +91,16 @@ private DaprClient buildDaprClientHttp() {
...
-        if (daprHttClient == null) {
-            synchronized (okHttpClient) {
-                if (daprHttClient == null) {
-                    DaprHttp daprHtt = new DaprHttp(Constants.DEFAULT_HTTP_PROTOCOL_IDENTIFIED+host, port, okHttpClient);
-                    daprHttClient = new DaprClientHttpAdapter(daprHtt);
+        if (this.daprHttClient == null) {
+            synchronized (DaprClientBuilder.class) {
+                if (this.daprHttClient == null) {
+                    OkHttpClient okHttpClient = new OkHttpClient.Builder().build();
+                    DaprHttp daprHtt = new DaprHttp(Constants.DEFAULT_BASE_HTTP_URL, port, okHttpClient);
```
Comment (AndresRoblesMX) [inline sdk/src/main/java/io/dapr/client/DaprClientBuilder.java]: Here was the only play where the variable host was actually used, with this approach we can remove that variable all togheter along with the validation for it.

**Comment by artursouza** [comment]
Addressed comments. Merging.

**Pull request merged**